src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunCodegenNodeFactory.cs

```diff
@@ -729,5 +723,56 @@ protected override IMethodNode CreateUnboxingStubNode(MethodDesc method)
         {
             throw new NotImplementedException();
         }
+
+        struct TypeAndMethod
```

Comment:
Not implementing `IEquatable<TypeAndMethod>` will lead to a lot of boxing when this is used in a dictionary.

Reply:
Fixed in 3rd commit.